### PR TITLE
Fix some errors when HAVE_TBB is false

### PR DESCRIPTION
**Pull request by tobiasleibner**

### CODE REVIEW

**Comment by ftalbrecht** [comment]
Looks good to me, any objections @renemilk?

File: dune/stuff/grid/walker.hh [hunk 1]
```diff
@@ -185,7 +185,7 @@ public:
 
   void walk(const bool use_tbb = false)
   {
-#if DUNE_VERSION_NEWER(DUNE_COMMON, 3, 9) // EXADUNE
+#if DUNE_VERSION_NEWER(DUNE_COMMON, 3, 9) && HAVE_TBB // EXADUNE
```
Comment (renemilk) [inline dune/stuff/grid/walker.hh]: This should not be necessary. TBB is mandatory with dune-common in exadune.

**Comment by tobiasleibner** [comment]
I am using your DUNE-Multiscale-super module and thus the dune-common version that comes with it. For some reason, my clang builds do not have TBB (gcc builds do). With the changes in this pull request, everything works fine in the clang builds as long as threading.max_count is set to 1. So at least for my code it is possible to use that dune-common version without TBB.

**Comment by renemilk** [comment]
Alright, that's a little weird, but not worth investigating unless you want to use threading + clang. Would you mind tracking down the other occurrences of the guard as well? And if you're feeling really motivated substitute them with single ```DS_EXADUNE_BUILD``` macro that you'd add to config.h.cmake accordingly?

**Comment by tobiasleibner** [comment]
Couldn't find any other occurrences in dune-stuff. There are some in dune-gdt, but maybe just because I added them in my branch. Since there are only these two occurences I wasn't really motivated to add a special macro for them :).

**Comment by ftalbrecht** [comment]
Sounds good to me! Any objections against a merge, @renemilk?

**Comment by renemilk** [comment]
The ```GridWalker``` tests needs the same guards.

**Comment by tobiasleibner** [comment]
I changed dune/stuff/test/grid_walker.cc accordingly, is the GridWalker also tested elsewhere?

**Comment by renemilk** [comment]
Nope. Sorry, I've mixed up branches in review.

**Pull request merged**